### PR TITLE
Request places and addEditOperation test added

### PR DESCRIPTION
**Pull request by angry010101**
Added 2 test
- testAddEditOpToDb
- testRequestOpFromDb

### CODE REVIEW

File: opendb-core/src/main/java/org/openplacereviews/opendb/service/DBConsensusManager.java [hunk 1]
```diff
@@ -859,6 +860,8 @@ public OpBlockChain doInTransaction(TransactionStatus status) {
 			if(conn != null) {
 				try {
 					conn.close();
+					//FIXME
```
Comment (angry010101) [inline opendb-core/src/main/java/org/openplacereviews/opendb/service/DBConsensusManager.java]: Why we need this?
throw new SQLException();

File: opendb-core/src/main/java/org/openplacereviews/opendb/service/DBConsensusManager.java [hunk 2]
```diff
@@ -377,15 +377,16 @@ public OpObject getObjectById(String type, CompoundKey k, boolean content) throw
 				o[1] = type;
 				k.toArray(o, 2);
 				String table = dbSchema.getTableByType(type);
-				if (sz != dbSchema.getKeySizeByType(type)) {
-					return null;
-					//throw new UnsupportedOperationException();
+				if (sz > dbSchema.getKeySizeByType(type)) {
+					//return new OpObject(true);
```
Comment (vshcherb) [inline opendb-core/src/main/java/org/openplacereviews/opendb/service/DBConsensusManager.java]: delete comments

File: opendb-core/src/main/java/org/openplacereviews/opendb/service/DBConsensusManager.java (hunk 1 above)
Comment (vshcherb) [reply opendb-core/src/main/java/org/openplacereviews/opendb/service/DBConsensusManager.java]: It's not needed you've just added an unclear comment.

I'm also not sure if connection close is really needed here, probably to make sure that  transaction is closed
but in test it's probably unnecessary

File: opendb-core/src/main/java/org/openplacereviews/opendb/service/SettingsManager.java [hunk 3]
```diff
@@ -522,7 +522,7 @@ protected String convertToDBValue(Map<String, Object> obj) {
 	// BLOCK AND HISTORY
 	public final CommonPreference<Boolean> OPENDB_STORE_HISTORY = registerBooleanPreference("opendb.db.store-history", true, "Store history of operations").editable().restartNeeded();
 	public final CommonPreference<Double> OPENDB_COMPACT_COEFICIENT = registerDoublePreference("opendb.db.compactCoefficient", 1.0,  "Compact coefficient for compacting blockchain").editable();
-	public final CommonPreference<Integer> OPENDB_SUPERBLOCK_SIZE = registerIntPreference("opendb.db.dbSuperblockSize", 32,  "The amount of blocks to create superblock in a database").editable();
+	public final CommonPreference<Integer> OPENDB_SUPERBLOCK_SIZE = registerIntPreference("opendb.db.dbSuperblockSize", 2,  "The amount of blocks to create superblock in a database").editable();
```
Comment (vshcherb) [inline opendb-core/src/main/java/org/openplacereviews/opendb/service/SettingsManager.java]: Default values should not change

File: opendb-core/src/main/java/org/openplacereviews/opendb/api/ApiController.java [hunk 4]
```diff
@@ -128,7 +128,7 @@ public String logsList() throws FailedVerificationException {
 		public List<String> sblocks = new ArrayList<String>();
 	}
 
-	protected static class ObjectsResult {
```
Comment (vshcherb) [inline opendb-core/src/main/java/org/openplacereviews/opendb/api/ApiController.java]: why do we need to change this?

File: opendb-core/src/test/java/org/openplacereviews/opendb/service/OpBlockchainDbAccessTest.java [hunk 5]
```diff
@@ -90,12 +116,17 @@ public void beforeEachTest() throws SQLException, FailedVerificationException {
...
+		databaseBlockChain = dbConsensusManager.init(metadataDb);
 	}
 
 	@After
 	public void afterEachTest() throws Exception {
-		databaseServer.wipeDatabase();
+		try {
+			databaseServer.wipeDatabase();		
+		}
+		catch(Exception e) {
```
Comment (vshcherb) [inline opendb-core/src/test/java/org/openplacereviews/opendb/service/OpBlockchainDbAccessTest.java]: Formatting!

File: opendb-core/src/test/java/org/openplacereviews/opendb/service/OpBlockchainDbAccessTest.java [hunk 6]
```diff
@@ -162,4 +193,241 @@ public void testChangeToEqualParent() throws FailedVerificationException {
...
+	
+	private List<OpOperation> generateStartOpForTest(OpBlockChain blc) throws FailedVerificationException {
+		JsonFormatter jf = new JsonFormatter();
+
+		String version4 = "{\"id\": [\"9G2GCG\",\"wlkomu\"],\"source\": {\"osm\": [{\"id\": 43383147,\"lat\": 50.44494543216025,\"lon\": 30.508777081059165,\"tags\": {\"name\": \"Володимирський собор\",\"amenity\": \"place_of_worship\",\"name:de\": \"Wladimirkathedrale\",\"name:en\": \"St. Vladimir's Cathedral\",\"name:et\": \"Püha Volodõmõri katedraal\",\"name:fr\": \"Cathédrale Saint-Vladimir\",\"name:it\": \"Cattedrale di San Vladimiro\",\"name:nl\": \"Vladimirkathedraal\",\"name:ru\": \"Владимирский cобор\",\"name:uk\": \"Володимирський собор\",\"website\": \"http://www.katedral.org.ua/\",\"building\": \"cathedral\",\"religion\": \"christian\",\"wikidata\": \"Q1417441\",\"wikipedia\": \"uk:Володимирський собор (Київ)\",\"denomination\": \"ukrainian_orthodox\",\"addr:housenumber\": \"20\"},\"type\": \"way\",\"osm_tag\": \"amenity\",\"version\": \"15\",\"changeset\": \"75113080\",\"osm_value\": \"place_of_worship\",\"timestamp\": \"2019-09-30T17:17:23Z\"}],\"old-osm-ids\": []},\"version\": 4,\"placetype\": \"place_of_worship\"}";
```
Comment (vshcherb) [inline opendb-core/src/test/java/org/openplacereviews/opendb/service/OpBlockchainDbAccessTest.java]: Unnecessary content could be much shorter and better formatting

File: opendb-core/src/test/java/org/openplacereviews/opendb/service/OpBlockchainDbAccessTest.java [hunk 7]
```diff
@@ -162,4 +193,241 @@ public void testChangeToEqualParent() throws FailedVerificationException {
...
+		op5w.setSignedBy(serverName);
+		op5w.putObjectValue(OpOperation.F_CREATE, op5);
+		op5w.addCreated(op5);
+
+		opaw.setType(OP_ID);
+		opaw.setSignedBy(serverName);
+		opaw.putObjectValue(OpOperation.F_CREATE, opa);
+		opaw.addCreated(opa);
+		
+		blc.getRules().generateHashAndSign(opaw, serverKeyPair);
```
Comment (vshcherb) [inline opendb-core/src/test/java/org/openplacereviews/opendb/service/OpBlockchainDbAccessTest.java]: could be simplified a lot, too much copy/paste

**Pull request merged**